api/app/weather_models/machine_learning.py

```diff
@@ -170,6 +174,24 @@ def learn(self):
             self.session, self.model.id, self.station_code, start_date, self.max_learn_date)
         self.regression_models_v2.collect_data(query)
         self.regression_models_v2.train()
+        
+    def learn_precip_model(self, start_date):
```

Comment:
The last thing I'd consider is adding some tests for this function

Reply:
I incorporated testing into the machine learning tests of the other bias adjusted weather parameters in `test_machine_learning.py`.